notebooks/6_search_in_continuous_spaces/index.md

```diff
@@ -0,0 +1,282 @@
+
+[Local minima and saddle points	 PAGEREF _Toc86421110 \h 10](#_Toc86421110)
+
+[Vanishing and Exploding Gradients	 PAGEREF _Toc86421111 \h 10](#_Toc86421111)
+
+[**Conclusion	 PAGEREF _Toc86421112 \h 11**](#_Toc86421112)
+
+[**References	 PAGEREF _Toc86421113 \h 11**](#_Toc86421113)
+
+
+#
+#
+
+
+# Introduction
+In real life optimization problems, we mostly face continuous environments which are made of continuous variables such as time and location. Thus, in this lecture note we will discuss some useful search methods in continuous spaces. The typical problems are linear programming (LP) and diverse kinds of nonlinear programming (such as convex programming, quadratic programming or semidefinite programming). In this lecture note we will discuss different types of optimization techniques in continuous spaces, the differences between different types of optimization techniques, how convexity will help us solve optimization problems easier, difference between local and global optimization, gradient descent method and the challenges with finding the best learning rate.
+
+# Type of optimization techniques
+An essential step to optimization technique is to categorize the optimization model since the algorithms used for solving optimization problems are customized as per the nature of the problem. Various optimization problem types are: 
+## Linear programming (LP) and Nonlinear programming (NLP)
+Linear programming is a method to achieve the best outcome in a mathematical model whose requirements are represented by linear relationships whereas nonlinear programming is a process of solving an optimization problem where the constraints or the objective functions are nonlinear. Thus, this is the main difference between linear and nonlinear programming.
+## Continuous optimization and Integer programming (IP)
+In order to conceptualize the difference, linear programming can solve problems about minimizing (or maximizing) an objective function by continuous variables. For instance, maybe the optimal solution for a problem to be x1=5,46 and x2=2,65. But in integer programming we can only use integers as variables and we can’t have fractional numbers.
+## Constrained optimization and Unconstrained optimization 
+**Constrained optimization problems** consider the problem of optimizing an objective function subject to constraints on the variables. In general terms,
+
+minimize fx
+
+subject to    cix=0    ∀i ϵ ε
+
+cix≤0    ∀i ϵ τ 
+
+Where *f and* the functions ci(x) are all smooth, real-valued functions on a subset of Rn and ε and τ are index sets for equality and inequality constraints, respectively. The feasible set is the set of points x that satisfy the constraints.
+
+We denote the set of points for which all the constraints are satisfied as C, and say that any x ∈ C (resp. x ∈/ C) is feasible (resp. infeasible)
+
+***Unconstrained optimization problems*** the answers are constrained into being subject of set C as the picture bellow shows: 

```

Comment:
Try to avoid using different formats in a section. For example "Constrained optimization and Unconstrained optimization" is just bold while "Unconstrained optimization problems" is in bold and italics.